pkg/cloud/aws/services/cloudformation/bootstrap.go

```diff
@@ -203,6 +193,77 @@ func controllersPolicy(accountID string) *iam.PolicyDocument {
 					"StringLike": map[string]string{"iam:AWSServiceName": "elasticloadbalancing.amazonaws.com"},
 				},
 			},
+			{
+				Effect:   iam.EffectAllow,
+				Resource: iam.Resources{"*"},
+				Action: iam.Actions{
+					"ec2:CreateTags",
+					"ec2:RunInstances",
+					"elasticloadbalancing:CreateLoadBalancer",
+				},
+				Condition: iam.Conditions{
+					"StringEquals": map[string][]string{
+						fmt.Sprintf("aws:RequestTag/%s", tags.NameAWSProviderManaged): {"true"},
+						fmt.Sprintf("aws:RequestTag/%s", tags.NameAWSClusterAPIRole): {
+							tags.ValueAPIServerRole,
+							tags.ValueBastionRole,
+							tags.ValueCommonRole,
+							tags.ValuePrivateRole,
+							tags.ValuePublicRole,
+						},
+					},
+					"ForAllValues:StringEquals": map[string][]string{
+						"aws:TagKeys": {
+							tags.NameAWSProviderManaged,
+							tags.NameAWSClusterAPIRole,
+						},
+					},
+					"ForAllValues:StringLike": map[string][]string{
+						"aws:TagKeys": {
+							fmt.Sprintf("%s*", tags.NameKubernetesClusterPrefix),
+							tags.NameAWSClusterAPIRole,
+							tags.NameAWSProviderManaged,
+							"Name",
+						},
+					},
+				},
+			},
+			{
+				Effect: iam.EffectDeny,
+				Resource: iam.Resources{
+					"arn:aws:ec2:::instance/*",
```

Comment:
Similar to above, would this prevent us from reconciling AddtionalTags?

As a followup, maybe we shouldn't reconcile AdditionalTags and only apply on creation?

Reply:
I think it's prudent to not reconcile AdditionalTags if we really want to stop ourselves from deleting an instance not of our creation.

Reply:
Rewritten these with the intent that you can reconcile tags as long as the instance was already tagged with our management tags.